src/construction.cpp

```diff
@@ -1756,6 +1774,45 @@ void construct::done_mine_upstair( const tripoint_bub_ms &p, Character &player_c
     tmpmap.save();
 }
 
+void construct::done_dig_upslope( const tripoint_bub_ms &p, Character &player_character )
+{
+    map &here = get_map();
```

Comment:
This function also essentially duplicates the above.

Reply:
OK I did this but then I realized that it breaks the construction specials... I need help with that. Sorry for bothering you. :/

Reply:
How so?

Reply:
In `post_special_map = {{`, there is a list of construction specials. Merging the functions made the specials not refer to the correct strings anymore. And my C++ skills are too weak to solve this.

Reply:
Actually nevermind, I think it will work on a re-read of the relevant code. Will fix up the formatting etc later today, busy RN.